Traceback (most recent call last):
  File "/home/ryan/.conda/envs/newwork/lib/python3.9/site-packages/jupyter_cache/executors/utils.py", line 51, in single_nb_execution
    executenb(
  File "/home/ryan/.conda/envs/newwork/lib/python3.9/site-packages/nbclient/client.py", line 1204, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "/home/ryan/.conda/envs/newwork/lib/python3.9/site-packages/nbclient/util.py", line 84, in wrapped
    return just_run(coro(*args, **kwargs))
  File "/home/ryan/.conda/envs/newwork/lib/python3.9/site-packages/nbclient/util.py", line 62, in just_run
    return loop.run_until_complete(coro)
  File "/home/ryan/.conda/envs/newwork/lib/python3.9/asyncio/base_events.py", line 647, in run_until_complete
    return future.result()
  File "/home/ryan/.conda/envs/newwork/lib/python3.9/site-packages/nbclient/client.py", line 663, in async_execute
    await self.async_execute_cell(
  File "/home/ryan/.conda/envs/newwork/lib/python3.9/site-packages/nbclient/client.py", line 965, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/home/ryan/.conda/envs/newwork/lib/python3.9/site-packages/nbclient/client.py", line 862, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
h=100 # W/m/m/K
k=200 # W/m/K
R=1E-3# radius in m
L=60E-3# length in m
s=np.sqrt(2*h/k/R)
x=np.arange(0,70,10)*1e-3 # m
T= 20+80*(np.cosh(s*L-s*x)+h/s/k*np.sinh(s*L-s*x))/(np.cosh(s*L)+h/s/k*np.sinh(s*L))
------------------

[0;31m---------------------------------------------------------------------------[0m
[0;31mNameError[0m                                 Traceback (most recent call last)
Input [0;32mIn [1][0m, in [0;36m<cell line: 5>[0;34m()[0m
[1;32m      3[0m R[38;5;241m=[39m[38;5;241m1E-3[39m[38;5;66;03m# radius in m[39;00m
[1;32m      4[0m L[38;5;241m=[39m[38;5;241m60E-3[39m[38;5;66;03m# length in m[39;00m
[0;32m----> 5[0m s[38;5;241m=[39m[43mnp[49m[38;5;241m.[39msqrt([38;5;241m2[39m[38;5;241m*[39mh[38;5;241m/[39mk[38;5;241m/[39mR)
[1;32m      6[0m x[38;5;241m=[39mnp[38;5;241m.[39marange([38;5;241m0[39m,[38;5;241m70[39m,[38;5;241m10[39m)[38;5;241m*[39m[38;5;241m1e-3[39m [38;5;66;03m# m[39;00m
[1;32m      7[0m T[38;5;241m=[39m [38;5;241m20[39m[38;5;241m+[39m[38;5;241m80[39m[38;5;241m*[39m(np[38;5;241m.[39mcosh(s[38;5;241m*[39mL[38;5;241m-[39ms[38;5;241m*[39mx)[38;5;241m+[39mh[38;5;241m/[39ms[38;5;241m/[39mk[38;5;241m*[39mnp[38;5;241m.[39msinh(s[38;5;241m*[39mL[38;5;241m-[39ms[38;5;241m*[39mx))[38;5;241m/[39m(np[38;5;241m.[39mcosh(s[38;5;241m*[39mL)[38;5;241m+[39mh[38;5;241m/[39ms[38;5;241m/[39mk[38;5;241m*[39mnp[38;5;241m.[39msinh(s[38;5;241m*[39mL))

[0;31mNameError[0m: name 'np' is not defined
NameError: name 'np' is not defined

